src/main/java/mate/academy/bookstore/repository/book/BookSpecificationProviderManager.java

```diff
@@ -0,0 +1,25 @@
+package mate.academy.bookstore.repository.book;
+
+import java.util.List;
+import lombok.RequiredArgsConstructor;
+import mate.academy.bookstore.model.Book;
+import mate.academy.bookstore.repository.SpecificationProvider;
+import mate.academy.bookstore.repository.SpecificationProviderManager;
+import org.springframework.stereotype.Component;
+
+@RequiredArgsConstructor
+@Component
+public class BookSpecificationProviderManager implements SpecificationProviderManager<Book> {
+    private static final String CANNOT_FIND_SPEC_PROVIDER_FOR_KEY
+            = "Can't find correct specification provider for key: ";
+    private final List<SpecificationProvider<Book>> specificationProviderList;
+
+    @Override
+    public SpecificationProvider<Book> getSpecificationProvider(String key) {
+        return specificationProviderList.stream()
+                .filter(k -> k.getKey().equals(key))
```

Comment:
It's better to use the first letter of entity name that you process. 
You process specificationProvider, so it is better to use s(specification) or p(provider) rather than k.